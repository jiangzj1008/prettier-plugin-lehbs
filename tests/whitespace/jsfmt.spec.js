run_spec(__dirname, { printWidth: 40 });
