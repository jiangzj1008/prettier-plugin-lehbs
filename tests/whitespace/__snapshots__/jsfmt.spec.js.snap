// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basics.hbs 1`] = `
{{!-- after --}}
<span>
    foo <span>bar</span>
</span>

{{!-- before --}}
<span>
  <span>bar</span> foo
</span>

{{!-- within --}}
<span>
  foo <span> bar </span>
</span>

{{!-- break components --}}
<div>
  <SuperSelect>
    <p>foo<span>bar bar bar</span></p><h1><span><em>yep</em></span></h1>
  </SuperSelect>
  <h2>nope</h2>
</div>

<div>
   hello <strong>hi</strong> <foo>sdkflsdfjk</foo>
</div>;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{! after }}
<span>
  foo
  <span>bar</span>
</span>

{{! before }}
<span>
  <span>bar</span>
  foo
</span>

{{! within }}
<span>
  foo
  <span> bar </span>
</span>

{{! break components }}
<div>
  <SuperSelect>
    <p>foo<span>bar bar bar</span></p><h1
    ><span><em>yep</em></span></h1>
  </SuperSelect>
  <h2>nope</h2>
</div>

<div>
  hello
  <strong>hi</strong>
  <foo>sdkflsdfjk</foo>
</div>;
`;

exports[`boss.hbs 1`] = `
<p>Hi  {{firstName}}  {{lastName}}   , welcome!</p>
{{#component propA}}
    for {{propB}}  do {{propC}} f
{{/component}}
{{#component propA}}
    for {{propB}}  <span>name</span>do {{propC}} f
{{/component}}
{{propA}} {{propB}}
{{propC}}{{propD}}
<span>{{propE}} {{propF}}</span>
<span>{{propG}}{{propH}}</span>

  

hey
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<p>Hi
  {{firstName}}
  {{lastName}}
  , welcome!</p>
{{#component propA}}
  for
  {{propB}}
  do
  {{propC}}
  f
{{/component}}
{{#component propA}}
  for
  {{propB}}
  <span>name</span>do
  {{propC}}
  f
{{/component}}
{{propA}}
{{propB}}
{{propC}}{{propD}}
<span>{{propE}} {{propF}}</span>
<span>{{propG}}{{propH}}</span>

hey
`;

exports[`curly.hbs 1`] = `
<p>Your username is @{{name}}</p>
<p>Hi {{firstName}} {{lastName}}</p>~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<p>Your username is @{{name}}</p>
<p>Hi {{firstName}} {{lastName}}</p>
`;

exports[`display-inline-block.hbs 1`] = `
<button>Click here! Click here! Click here! Click here! Click here! Click here!</button>
<button>
Click here! Click here! Click here! Click here! Click here! Click here!
</button>
<div>
<button>Click here! Click here! Click here! Click here! Click here! Click here!</button><button>Click here! Click here! Click here! Click here! Click here! Click here!</button>
</div>
<div>
<button>Click here! Click here! Click here! Click here! Click here! Click here!</button>
<button>Click here! Click here! Click here! Click here! Click here! Click here!</button>
</div>
<video src="brave.webm"></video>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<button>Click here! Click here! Click
  here! Click here! Click here! Click
  here!</button>
<button>
  Click here! Click here! Click here!
  Click here! Click here! Click here!
</button>
<div>
  <button>Click here! Click here! Click
    here! Click here! Click here! Click
    here!</button><button>Click here!
    Click here! Click here! Click here!
    Click here! Click here!</button>
</div>
<div>
  <button>Click here! Click here! Click
    here! Click here! Click here! Click
    here!</button>
  <button>Click here! Click here! Click
    here! Click here! Click here! Click
    here!</button>
</div>
<video src="brave.webm"></video>
`;

exports[`display-none.hbs 1`] = `
{{!-- TO FIX --}}
<HEAD><META CHARSET="utf-8" /><TITLE>My tITlE</TITLE><META NAME="description" content="My CoNtEnT" /></HEAD>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{! TO FIX }}
<HEAD><META CHARSET="utf-8" /><TITLE>My
    tITlE</TITLE><META
    NAME="description"
    content="My CoNtEnT"
  /></HEAD>
`;

exports[`fill.hbs 1`] = `
<p>
  <img
    src="/images/pansies.jpg"
    alt="about fedco bottom image"
    style="float: left;"
  /><strong>We are a cooperative</strong>, one of the few seed companies so organized
  in the United States. Because we do not have an individual owner or beneficiary,
  profit is not our primary goal. Consumers own 60% of the cooperative and worker
  members 40%. Consumer and worker members share proportionately in the cooperative&#8217;s
  profits through our annual patronage dividends.
</p>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<p>
  <img
    src="/images/pansies.jpg"
    alt="about fedco bottom image"
    style="float: left;"
  /><strong>We are a cooperative</strong>,
  one of the few seed companies so
  organized in the United States.
  Because we do not have an individual
  owner or beneficiary, profit is not
  our primary goal. Consumers own 60% of
  the cooperative and worker members
  40%. Consumer and worker members share
  proportionately in the
  cooperative&#8217;s profits through
  our annual patronage dividends.
</p>
`;

exports[`inline-element.hbs 1`] = `
{{!-- TO FIX --}}
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce cursus massa vel augue 
vestibulum facilisis in porta turpis. Ut faucibus lectus sit amet urna consectetur dignissim.
Sam vitae neque quis ex dapibus faucibus at sed ligula. Nulla sit amet aliquet nibh.
Vestibulum at congue mi. Suspendisse vitae odio vitae massa hendrerit mattis sed eget dui.
Sed eu scelerisque neque. Donec <b>maximus</b> rhoncus pellentesque. Aenean purus turpis, vehicula 
euismod ante vel, ultricies eleifend dui. Class aptent taciti sociosqu ad litora torquent per 
conubia nostra, per inceptos himenaeos. Donec in ornare velit.</p>

<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce cursus massa vel augue 
vestibulum facilisis in porta turpis. Ut faucibus lectus sit amet urna consectetur dignissim.
Sam vitae neque quis ex dapibus faucibus at sed ligula. Nulla sit amet aliquet nibh.
Vestibulum at congue mi. Suspendisse vitae odio vitae massa hendrerit mattis sed eget dui.
Sed eu scelerisque neque. Donec <a href="#"><b>maximus</b></a> rhoncus pellentesque. Aenean purus turpis, vehicula 
euismod ante vel, ultricies eleifend dui. Class aptent taciti sociosqu ad litora torquent per 
conubia nostra, per inceptos himenaeos. Donec in ornare velit.</p>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{! TO FIX }}
<p>Lorem ipsum dolor sit amet,
  consectetur adipiscing elit. Fusce
  cursus massa vel augue vestibulum
  facilisis in porta turpis. Ut faucibus
  lectus sit amet urna consectetur
  dignissim. Sam vitae neque quis ex
  dapibus faucibus at sed ligula. Nulla
  sit amet aliquet nibh. Vestibulum at
  congue mi. Suspendisse vitae odio
  vitae massa hendrerit mattis sed eget
  dui. Sed eu scelerisque neque. Donec
  <b>maximus</b>
  rhoncus pellentesque. Aenean purus
  turpis, vehicula euismod ante vel,
  ultricies eleifend dui. Class aptent
  taciti sociosqu ad litora torquent per
  conubia nostra, per inceptos
  himenaeos. Donec in ornare velit.</p>

<p>Lorem ipsum dolor sit amet,
  consectetur adipiscing elit. Fusce
  cursus massa vel augue vestibulum
  facilisis in porta turpis. Ut faucibus
  lectus sit amet urna consectetur
  dignissim. Sam vitae neque quis ex
  dapibus faucibus at sed ligula. Nulla
  sit amet aliquet nibh. Vestibulum at
  congue mi. Suspendisse vitae odio
  vitae massa hendrerit mattis sed eget
  dui. Sed eu scelerisque neque. Donec
  <a href="#"><b>maximus</b></a>
  rhoncus pellentesque. Aenean purus
  turpis, vehicula euismod ante vel,
  ultricies eleifend dui. Class aptent
  taciti sociosqu ad litora torquent per
  conubia nostra, per inceptos
  himenaeos. Donec in ornare velit.</p>
`;

exports[`non-breaking-whitespace.hbs 1`] = `
<!-- normal whitespaces -->
<span>Nihil aut odit omnis. Quam maxime est molestiae. Maxime dolorem dolores voluptas quaerat ut qui sunt vitae error.</span>
<!-- non-breaking whitespaces -->
<span>Nihil aut odit omnis. Quam maxime est molestiae. Maxime dolorem dolores voluptas quaerat ut qui sunt vitae error.</span>
<!-- non-breaking narrow whitespaces -->
<span>Prix : 32 €</span>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!-- normal whitespaces -->
<span>Nihil aut odit omnis. Quam maxime
  est molestiae. Maxime dolorem dolores
  voluptas quaerat ut qui sunt vitae
  error.</span>
<!-- non-breaking whitespaces -->
<span
>Nihil aut odit omnis. Quam maxime est molestiae. Maxime dolorem dolores voluptas quaerat ut qui sunt vitae error.</span>
<!-- non-breaking narrow whitespaces -->
<span>Prix : 32 €</span>
`;

exports[`preserved-spaces-and-breaks.hbs 1`] = `
<SomeComponent />
{{name}}



Some sentence with  {{dynamic}}  expressions.

sometimes{{nogaps}}areimportant<Hello></Hello>


{{name}}  is your name


{{#block}}
{{/block}}


{{#block}}
  some {{text}}
{{/block}}

{{#block}}

  some {{text}}
{{/block}}

<HelloWorld>
</HelloWorld>

<HelloWorld>
  some {{text}}
</HelloWorld>

<HelloWorld>

  some {{text}}
</HelloWorld>

<div class="a list of classes">
</div>

<div class="a list of classes">

</div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<SomeComponent />
{{name}}

Some sentence with
{{dynamic}}
expressions. sometimes{{nogaps}}areimportant<Hello
/>

{{name}}
is your name

{{#block}}{{/block}}

{{#block}}
  some
  {{text}}
{{/block}}

{{#block}}

  some
  {{text}}
{{/block}}

<HelloWorld />

<HelloWorld>
  some
  {{text}}
</HelloWorld>

<HelloWorld>

  some
  {{text}}
</HelloWorld>

<div class="a list of classes">
</div>

<div class="a list of classes">

</div>
`;

exports[`punctuation.hbs 1`] = `
<span>This is your name: {{name}}.</span>
<span>This is your name: {{name}} (employee)</span>
<span>This is your name: {{name}} ({{role}})</span>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<span>This is your name:
  {{name}}.</span>
<span>This is your name:
  {{name}}
  (employee)</span>
<span>This is your name:
  {{name}}
  ({{role}})</span>
`;

exports[`surrounding-linebreak.hbs 1`] = `
<span>123</span>
<span>
123</span>
<span>123
</span>
<span>
123
</span>

<div>123</div>
<div>
123</div>
<div>123
</div>
<div>
123
</div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<span>123</span>
<span>
  123</span>
<span>123
</span>
<span>
  123
</span>

<div>123</div>
<div>
  123</div>
<div>123
</div>
<div>
  123
</div>
`;

exports[`table.hbs 1`] = `
<table>
  <thead>
    <tr>
      <th>A</th>
      <th>B</th>
      <th>C</th>
    </tr>
  </thead>
</table>

<table><thead><tr><th>A</th><th>B</th><th>C</th></tr></thead></table>

<table> <thead> <tr> <th> A </th> <th> B </th> <th> C </th> </tr> </thead> </table>

<table>
  <thead>
    <tr>
    </tr>
  </thead>
</table>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<table>
  <thead>
    <tr>
      <th>A</th>
      <th>B</th>
      <th>C</th>
    </tr>
  </thead>
</table>

<table><thead><tr><th>A</th><th
      >B</th><th
      >C</th></tr></thead></table>

<table>
  <thead>
    <tr>
      <th> A </th>
      <th> B </th>
      <th> C </th>
    </tr>
  </thead>
</table>

<table>
  <thead>
    <tr>
    </tr>
  </thead>
</table>
`;

exports[`textarea.hbs 1`] = `
<Textarea />
{{#if true}}
  Test
{{/if}}

<Textarea />

{{#if true}}
  Test
{{/if}}

<Textarea />




{{#if true}}
  Test
{{/if}}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<Textarea />
{{#if true}}
  Test
{{/if}}

<Textarea />

{{#if true}}
  Test
{{/if}}

<Textarea />

{{#if true}}
  Test
{{/if}}
`;

exports[`whitespace-control.hbs 1`] = `
{{~#if foo~}}
  abc
{{~/if~}}

{{~#if foo}}
  abc
{{/if~}}

{{foo~}}

{{~foo abc}}

{{~#if bar}}
  if1
{{~else~}}
  else
{{~/if~}}

{{~#if bar}}
  if1
{{~else if foo~}}
  else if foo
{{else~}}
  else
{{/if~}}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{~#if foo~}}
  abc
{{~/if~}}

{{~#if foo}}
  abc
{{/if~}}

{{foo~}}

{{~foo abc}}

{{~#if bar}}
  if1
{{~else~}}
  else
{{~/if~}}

{{~#if bar}}
  if1
{{~else if foo~}}
  else if foo
{{else~}}
  else
{{/if~}}
`;
